### PR TITLE
feat: Track request with headers

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/axiosWrappers.ts [hunk 1]
```diff
@@ -6,15 +6,25 @@ import axios, {
...
 import { transformDateInRequest, transformDateInResponse } from './utils';
 
 /** @hidden */
-export function generateAxiosInstance(baseUrl: string): AxiosInstance {
+export function generateAxiosInstance(
+  baseUrl: string,
+  appId?: string
+): AxiosInstance {
+  const headers: { [key: string]: string } = {
+    'x-cdp-sdk': `CogniteJavascriptSDK:${packageJson.version}`,
```
Comment (polomani) [inline src/axiosWrappers.ts]:
```suggestion
    'x-cdp-sdk': `CogniteJavaScriptSDK:${packageJson.version}`,
```

File: src/__tests__/axiosWrappers.spec.ts [hunk 2]
```diff
@@ -42,6 +43,31 @@ describe('axiosWrappers', () => {
...
   });
 
+  describe('x-cdp headers', () => {
+    test('x-cdp-sdk', async () => {
+      responseMock.mockReturnValueOnce([200, {}]);
+      await rawRequest(instance, { method: 'get', url });
+      expect(responseMock).toHaveBeenCalledTimes(1);
+      const config = responseMock.mock.calls[0][0] as AxiosRequestConfig;
+      expect(config.headers['x-cdp-sdk']).toBeDefined();
+      expect(config.headers['x-cdp-sdk']).toMatch(/^CogniteJavascriptSDK:.+$/);
```
Comment (polomani) [inline src/__tests__/axiosWrappers.spec.ts]:
```suggestion
      expect(config.headers['x-cdp-sdk']).toMatch(/^CogniteJavaScriptSDK:.+$/);
```

File: src/axiosWrappers.ts [hunk 3]
```diff
@@ -6,15 +6,25 @@ import axios, {
...
-export function generateAxiosInstance(baseUrl: string): AxiosInstance {
+export function generateAxiosInstance(
+  baseUrl: string,
+  appId?: string
+): AxiosInstance {
+  const headers: { [key: string]: string } = {
+    'x-cdp-sdk': `CogniteJavaScriptSDK:${packageJson.version}`,
+  };
+  if (appId) {
+    headers['x-cdp-app'] = appId;
```
Comment (polomani) [inline src/axiosWrappers.ts:23]: isn't this one has to be called `x-cdp-application`?

Comment (ghost) [reply src/axiosWrappers.ts:23]: It should be `x-cdp-app`

**Review by polomani — APPROVED** [review]

**Pull request merged**